### DISCUSSION TITLE
help for new integration

### DISCUSSION THREAD

**Discussion by cabriltt** [djansen1987/SAJeSolar#10] [Q&A] [answered]
Hi
I have just instaled a new SAJ h1 series in my home and i am trying to integrate in home assinstant with no luck.
I have a meter but i dont think i have a sec module.
After the configuration al the sensors have the state as unkown.
Can you help me?
Many thanks

**Comment by djansen1987** [comment]
Hi @cabriltt,

Sorry for the late response, i was on holidays.

is this the device you have ? [https://nl.saj-electric.com/product/info-ce314562ed6c4467bde57cfbb31e6f95.html](https://nl.saj-electric.com/product/info-ce314562ed6c4467bde57cfbb31e6f95.html)
If so this custom_component was not created for this inverter at first. It could be the api's that are currently used for the other inverters have overlapping endpoints. If not the best way to support it to get login details to an account on which the device is configured. Also in creating such a integration goes a lot of time and effort on which i cannot make promises and would al be done on best effort. It would be nice to visit the donation page afterward if i would get it to work but the is always up to you.

I have done this before for the SEC module on which i had no access and another user provided credentials to his plant. Getting this right cost me a couple of weeks worth of evenings reversing the api.
If your you are okay with this you can send the login to github-h1@djansen.nl i will promise to be careful with your details as i only like this feature to become available to more people. 

Another option would be to share all the API endpoints you can find form the website and there value positions you need from it. Have to warn this will involve lot's of communication and trial and error which will take a lot more time.

Getting api's details can be done via the chrome developer tools under netwerk. Get details via reponse tap of preview below screenshot:

![image](https://user-images.githubusercontent.com/20375973/137197383-984c802a-90f9-4ea5-9758-a49a3139b7b7.png)

**Reply by djansen1987** [reply]
Received e-mail, thanks

**Comment by djansen1987** [comment, marked as answer]
Support should be added in the latest update. Please let me know if it works for you.

details: 
[https://github.com/djansen1987/SAJeSolar/releases/tag/v1.2.5](https://github.com/djansen1987/SAJeSolar/releases/tag/v1.2.5)

**Reply by djansen1987** [reply]
Successfully tested by user. Closing discussion
